### PR TITLE
Add unit test to check `app.booting` flag

### PR DESCRIPTION
**Pull request by superkhau**
Connected to strongloop/loopback-boot#77

### CODE REVIEW

**Comment by superkhau** [comment]
@bajtos Review please.

File: test/executor.test.js [hunk 1]
```diff
@@ -53,11 +53,17 @@ describe('executor', function() {
...
-    it('should set the booting status', function(done) {
+    it('should set the `booting` flag during execution', function(done) {
       expect(app.booting).to.be.undefined();
-      boot.execute(app, dummyInstructions, function(err) {
+
+      boot.execute(app, simpleAppInstructions(), function(err) {
         expect(err).to.be.undefined();
+
+        var bootingFlagSet = process.bootFlags.indexOf('booting') > -1;
+        expect(bootingFlagSet).to.be.ok();
```
Comment (bajtos) [inline test/executor.test.js]: `expect(process.bootFlags).to.include('booting')`

Comment (bajtos) [inline test/executor.test.js]: Please consider learning yourself more familiar with the different assertions provided by Chai, see http://chaijs.com/api/bdd/. The benefit of these assertions (compared to your solution above) is a much more helpful error message reported when the check fails.

Comment (superkhau) [inline test/executor.test.js]: Will do. ;) Thanks for the suggestion.

**Comment by superkhau** [comment]
@bajtos Updated based on code review comments. Good to merge?

**Comment by superkhau** [comment]
@slnode test please

**Comment by bajtos** [comment]
There's one more thing I don't like about the current approach, sorry for realising it too late. By reusing `bootFlags`, you now have to change other unrelated tests, plus these other tests are now tightly coupled with the order in which the newly added boot script is executed. This will make the maintenance more difficult.

Please rework the test to use different (new) flag, so that you don't have to modify any existing tests in this PR.

**Comment by bajtos** [comment]
And also please rebase on top of the current master to pull in the fix for the browserify-related test failures.

**Comment by superkhau** [comment]
@bajtos Reworked. Review again please.

**Comment by superkhau** [comment]
@slnode test please

File: test/executor.test.js [hunk 2]
```diff
@@ -53,11 +53,12 @@ describe('executor', function() {
...
   describe('when booting', function() {
-    it('should set the booting status', function(done) {
+    it('should set the `booting` flag during execution', function(done) {
       expect(app.booting).to.be.undefined();
-      boot.execute(app, dummyInstructions, function(err) {
+      boot.execute(app, simpleAppInstructions(), function(err) {
         expect(err).to.be.undefined();
-        expect(app.booting).to.be.false();
+        expect(process.bootingFlagSet).to.be.ok();
+        expect(app.booting).to.not.be.ok();
```
Comment (bajtos) [inline test/executor.test.js]: I am curious, why did you have to change `.to.be.false()` to `.to.not.be.ok()`?

**Comment by bajtos** [comment]
LGTM.

File: test/executor.test.js (hunk 2 above)
Comment (superkhau) [inline test/executor.test.js]: That was by accident actually. I looked up `ok()` and `...not.be.ok()` and `true()`/`false()` makes more sense but I forgot to change it back. I will submit another PR to change it back before we forget about it.

**Pull request merged**